shared/js/background/chrome-events.es6.js

```diff
@@ -262,6 +262,31 @@ chrome.webRequest.onBeforeSendHeaders.addListener(
     ['blocking', 'requestHeaders']
 )
 
+/**
+ * Global Privacy Control
+ */
+const GPC = require('./GPC.es6')
+
+// Set GPC property on DOM if enabled.
+chrome.webNavigation.onCommitted.addListener(details => {
+    GPC.injectDOMSignal(details.tabId)
+})
+
+// Attach GPC header to all requests if enabled.
+chrome.webRequest.onBeforeSendHeaders.addListener(
+    request => {
+        const GPCHeader = GPC.getHeader()
+
+        if (GPCHeader) {
+            let requestHeaders = request.requestHeaders
+            requestHeaders.push(GPCHeader)
+            return {requestHeaders: requestHeaders}
+        }
+    },
+    {urls: ['<all_urls>']},
+    ['blocking', 'requestHeaders']
+)
+
```

Comment:
for majority of users those two listeners will do nothing, it'd be better to wrap them in `if (settings.getSetting('GPCEnabled')) {` but I understand that there is no clean way for us to start/stop those listeners whenever the setting changes?

Reply:
Definitely the case for the header, but to match the most recent spec we have to inject a DOM signal regardless of whether setting is on or not--`navigator.globalPrivacyControl` will be true if enabled, and false otherwise.

I guess we could potentially wrap the header listener in an if statement, then add some logic to the settings toggle to add/remove the listener on change. What do you think, worth doing?

Reply:
Ah good point about the DOM prop. Yeah, going after the header is probably not worth the effort.